Assets/HoloToolkit/Utilities/Scripts/WorldAnchorManager.cs

```diff
@@ -0,0 +1,216 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+using UnityEngine.VR.WSA.Persistence;
+using System.Collections.Generic;
+using UnityEngine.VR.WSA;
+
+namespace HoloToolkit.Unity
+{
+    /// <summary>
+    /// Wrapper around world anchor store to streamline some of the
+    /// persistence api busy work.
+    /// </summary>
+    public class WorldAnchorManager : Singleton<WorldAnchorManager>
+    {
+        /// <summary>
+        /// To prevent initializing too many anchors at once
+        /// and to allow for the WorldAnchorStore to load asyncronously
+        /// without callers handling the case where the store isn't loaded yet
+        /// we'll setup a queue of anchor attachment operations.  
+        /// The AnchorAttachmentInfo struct has the data needed to do this.
+        /// </summary>
+        private struct AnchorAttachmentInfo
+        {
+            public GameObject gameObjectToAnchor { get; set; }
```

Comment:
NIT: Public properties are PascalCased. 
gameObjectToAnchor should be GameObjectToAnchor.